client/lib/purchases/index.ts

```diff
@@ -100,32 +113,34 @@ function getSubscriptionsBySite( subscriptions ) {
 		.sort( ( a, b ) => ( a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1 ) );
 }
 
-function getName( purchase ) {
+export function getName( purchase: Purchase ): string {
 	if ( isDomainRegistration( purchase ) || isDomainMapping( purchase ) ) {
-		return purchase.meta;
+		return purchase.meta ?? '';
```

Comment:
Domain products should always have a `meta` value, but the type does not guarantee that so just in case they don't, this will still return a string rather than `undefined`. [This call-site](https://github.com/Automattic/wp-calypso/blob/9c191e6ac24fe868cb6ec47f262945642f52e6fc/client/components/marketing-survey/cancel-jetpack-form/index.tsx#L380) of the function assumes it always returns a string, and since the fallback seems very unlikely I decided to make it always a string.